addons/account/tests/test_account_move_reconcile.py

```diff
@@ -5874,3 +5874,57 @@ def test_partial_reconcile_amounts_of_several_matching_lines(self):
             {'amount': 1001.0, 'debit_move_id': line_2.id, 'credit_move_id': line_5.id},
             {'amount': 1002.0, 'debit_move_id': line_3.id, 'credit_move_id': line_4.id},
         ])
+
+    def test_reconcile_cash_basis_payment_term_full_amount(self):
+        """ Test cash basis accounting with a payment term with multiple installments but paying full amount.
+        When creating a payment for the full amount instead of the first installment,
+        the cash basis entries should have proportional tax amounts, not full tax amounts.
+        """
+        self.env.company.tax_exigibility = True
+
+        # Create payment term: 30% now, balance 60 days
+        payment_term = self.env['account.payment.term'].create({
+            'name': '30% now, balance 60 days',
+            'line_ids': [
+                Command.create({
+                    'value': 'percent',
+                    'value_amount': 30.0,
+                    'nb_days': 0,
+                }),
+                Command.create({
+                    'value': 'percent',
+                    'value_amount': 70.0,
+                    'nb_days': 60,
+                }),
+            ],
+        })
+
+        # Create invoice with cash basis tax and payment term
+        invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_a.id,
+            'invoice_date': '2016-01-01',
+            'date': '2016-01-01',
+            'invoice_payment_term_id': payment_term.id,
+            'invoice_line_ids': [Command.create({
+                'product_id': self.product_a.id,
+                'price_unit': 100.0,
```

Comment:
Put a price at 99.99 here and 2 payment terms at 50.
You would have 2 tax amounts of 16.67 while the total tax amount should be 33.33.
That's the reason we have to put the residual and not just use the percentage.